R/workflow.R

```diff
@@ -0,0 +1,191 @@
+#' \item{plotTimeProfile()}{Plot time profile, residuals and their distribution}
+#' \item{plotPKParameters()}{Plot PK parameters}
+#' \item{plotSensitivity()}{Plot sensitivity analysis results}
+#' \item{runWorkflow()}{Run the active tasks of population worklfow}
+#' }
+#' @export
+#' @import tlf
+#' @import ospsuite
+#' @format NULL
+PopulationWorkflow <- R6::R6Class(
+  "PopulationWorkflow",
+  inherit = Workflow,
+  
+  public = list(
+    populationSimulation = NULL,
+    pkParametersCalculation = NULL,
+    sensitivityAnalysis = NULL,
+    demographyPlot = NULL,
+    timeProfilePlot = NULL,
+    pkParametersPlot = NULL,
+    sensitivityPlot =NULL,
+   
+    initialize = function(...){
+      super$initialize(...)
+      
+      self$setPopulationSimulationSettings()
+      self$setPKParametersCalculationSettings()
+      self$setSensitivityAnalysisSettings()
+      
+      self$setDemographyPlotSettings()
+      self$setTimeProfilePlotSettings()
+      self$setpkParametersPlotSettings()
+      self$setSensitivityPlotSettings()
+    },
+    
+    setPopulationSimulationSettings = function(populationsList = NULL,
+                                               active = TRUE,
+                                               message = NULL){
+      self$populationSimulation <- Task$new(input = populationsList %||% self$population,
+                                            active = active,
+                                            message = message %||% "Simulate population")
+    },
+    setPKParametersCalculationSettings = function(message = NULL){
+      self$pkParametersCalculation <- Task$new(message = message %||% "Calculate PK parameters")
+    },
+    setSensitivityAnalysisSettings = function(message = NULL){
+      self$sensitivityAnalysis <- Task$new(message = message %||% "Analyze sensitivity")
+    },
+    setDemographyPlotSettings = function(message = NULL){
+      self$demographyPlot <- Task$new(message = message %||% "Plot demography")
+    },
+    setTimeProfilePlotSettings = function(message = NULL){
+      self$timeProfilePlot <- Task$new(message = message %||% "Plot time profile")
+    },
+    setpkParametersPlotSettings = function(message = NULL){
+      self$pkParametersPlot <- Task$new(message = message %||% "Plot PK parameters")
+    },
+    setSensitivityPlotSettings = function(message = NULL){
+      self$sensitivityPlot <- Task$new(message = message %||% "Plot sensitivity analysis")
+    },
+    
+    runPopulationSimulation = function(){
+      populationResults <- runSimulation(self$models, self$population)
+      self$populationSimulation$output <- populationResults
+      save(populationResults, file = file.path(self$outputFolder, 'populationSimulation.Rdata'))
+      return(populationResults)
+    },
+    runPKParametersCalculation = function(){},
+    runSensitivityAnalysis = function(){},
+    
+    plotDemography = function(parameterNames = c(StandardPath$Age, StandardPath$Weight, StandardPath$Height)){

```

Comment:
@pchelle For exampel this function does not use anything from the workflow aside from the simulation and the population
so we can imagine a sile function called plotDemography that takes `simulation, population, input, output` and does it's work

The args `simulation, population, input, output` will probably be a recurring thing to use . we will see soon enough